backend/copr_backend/storage.py

```diff
@@ -216,42 +217,63 @@ def init_project(self, dirname, chroot):
         response = self.client.create_publication(repository)
         return response.ok
 
+    def upload_rpm(self, repository, path):
+        response = self.client.create_content(repository, path)
+
+        if not response.ok:
+            self.log.error("Failed to create Pulp content for: %s, %s",
+                           path, response.text)
+            return
+
+        # This involves a lot of unnecessary waiting until every
+        # RPM content is created. Once we can reliably label Pulp
+        # content with Copr build ID, we should drop this code and stop
+        # creating the `pulp.json` file
+        task = response.json()["task"]
+        response = self.client.wait_for_finished_task(task)
+        return response
+
     def upload_build_results(self, chroot, results_dir, target_dir_name):
+        max_workers = min(32, (os.cpu_count() or 1) * 5) # This could probably be hardcoded
```

Comment:
This would be `min(32, 80)` in our case, and we can expect ~100+ processes uploading stuff to Pulp, i.e. 3200 parallel threads.